core/src/main/java/org/infinispan/functional/impl/MetaParamsInternalMetadata.java

```diff
@@ -31,8 +32,15 @@ public static Metadata from(MetaParams params) {
       return new MetaParamsInternalMetadata(params);
    }
 
-   MetaParamsInternalMetadata() {
+   @ProtoFactory
+   MetaParamsInternalMetadata(NumericVersion numericVersion, SimpleClusteredVersion clusteredVersion,
+                              long created, long lastUsed, long lifespan, long maxIdle) {
       this.params = new MetaParams(MetaParams.EMPTY_ARRAY, 0);
+      this.params.add(new MetaEntryVersion(numericVersion == null ? clusteredVersion : numericVersion));
+      if (created > -1) params.add(new MetaCreated(created));
```

Comment:
Intellij tells me it would look nicer if you used method refs in ```created()``` and ```lastUsed()```

Reply:
I was trying to only include the relevant changes, but I've updated now as it's only 2 lines.